### PR TITLE
add: enablers for docker environment

### PR DESCRIPTION
**Pull request by akaJSaunders**
Jira Ticket: https://companieshouse.atlassian.net/browse/SR-57

Styles requires CDN to be running, must be run with docker dev env

Instructions to run it
1. From suppression-web, run`DOCKER_BUILDKIT=1 docker build --ssh default -t suppression-web .`
2. Make sure you have latest pull of the `feature/add_suppression` branch of docker dev env
3. enable suppression and platform modules
4. `tilt up`

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -24,9 +24,7 @@ const env = nunjucks.configure([
 app.set('views', viewPath);
 app.set('view engine', 'njk');
 
-// serve static assets
-app.use('/static', express.static('dist/static'));
-env.addGlobal('CSS_URL', '/static/app.css');
+env.addGlobal('CDN_HOST', 'cdn.chs.local');
```
Comment (geek-soft) [inline src/app.ts:27]: That should be parametrised, at least at some point.

File: src/views/index.njk [hunk 2]
```diff
@@ -1,9 +1,18 @@
 {% extends "govuk/template.njk" %}
 
 {% block head %}
-    <link rel="stylesheet" type="text/css" media="all" href="{{ CSS_URL }}" />
+    <!--[if !IE 8]><!-->
+    <link href="//{{ CDN_HOST }}/stylesheets/govuk-frontend/v3.6.0/govuk-frontend-3.6.0.min.css" rel="stylesheet"/>
+    <link href="//{{ CDN_HOST }}/stylesheets/services/dissolution/application.css" rel="stylesheet"/>
```
Comment (geek-soft) [inline src/views/index.njk:6]: dissolution? nope!

Comment (geek-soft) [reply src/views/index.njk:6]: In fact this is not even needed here.

File: src/app.ts (hunk 1 above)
Comment (geek-soft) [reply src/app.ts:27]: Technically we can accept this as default for short time being until we have AWS env.

File: Makefile [hunk 3]
```diff
@@ -1,5 +1,5 @@
 .PHONY: build
-build: clean build-app build-static
+build: clean build-app
 	npm run build
```
Comment (geek-soft) [inline Makefile:3]: That line seems to be not needed since `build` task will call `clean` and `build-app`

File: Tiltfile.dev [hunk 4]
```diff
@@ -0,0 +1,27 @@
+local_resource(
+  name = 'dev:suppression-web:init',
+  cmd = 'npm --silent install'
+)
+
+local_resource(
+  name = 'dev:suppression-web:build',
```
Comment (geek-soft) [inline Tiltfile.dev:7]: This is not optimised for performance. It means that tiny change to NJK file will trigger compilation which is more time consuming than simple copy. Follow what appeals did.

File: gulpfile.js [hunk 5]
```diff
@@ -3,22 +3,19 @@ const sass = require("gulp-sass");
...
+gulp.task('copy-views', function () {
+  return gulp.src(paths.pages + '/**/*').pipe(gulp.dest(paths.build + '/views'));
 });
 
-// compiles the sass down to css
-gulp.task('sass', function() {
-  return gulp.src('./src/static/sass/*.scss')
-    .pipe(sass())
-    .pipe(gulp.dest('./dist/static'));
+gulp.task('copy-govukfrontend', function () {
```
Comment (geek-soft) [inline gulpfile.js:17]: Currently this runs on every file change which is pointless. Slim tasks down to focus only on things that are mandatory.

File: Tiltfile.dev [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  cmd = 'npm --silent run build',
+  deps = [
+    'src'
+  ]
+)
+
+custom_build(
+  ref = 'suppression-web:latest',
```
Comment (geek-soft) [inline Tiltfile.dev:15]:
```suggestion
  ref = '169942020521.dkr.ecr.eu-west-1.amazonaws.com/local/suppression-web:latest',
```

File: docker-compose.yaml [hunk 7]
```diff
@@ -0,0 +1,7 @@
+version: "3.8"
+
+services:
+  suppression-web:
+    image: 'suppression-web:latest'
```
Comment (geek-soft) [inline docker-compose.yaml:5]:
```suggestion
    image: '169942020521.dkr.ecr.eu-west-1.amazonaws.com/local/suppression-web:latest'
```

File: README.md [hunk 8]
```diff
@@ -26,8 +24,8 @@ Then open your browser and go to http://localhost:3000/suppress-my-details.
 cd to the root directory, and type the following commands:
 
 ```
-docker build -t tiny_rebels/suppression-web .
-docker run -p 3000:3000 -d tiny_rebels/suppression-web
+DOCKER_BUILDKIT=1 docker build --ssh default -t suppression-web .
```
Comment (geek-soft) [inline README.md:27]:
```suggestion
DOCKER_BUILDKIT=1 docker build --ssh default -t 169942020521.dkr.ecr.eu-west-1.amazonaws.com/local/suppression-web .
```

File: README.md [hunk 9]
```diff
@@ -26,8 +24,8 @@ Then open your browser and go to http://localhost:3000/suppress-my-details.
...
+docker run -p 3000:3000 -d suppression-web
```
Comment (geek-soft) [inline README.md:28]:
```suggestion
docker run -p 3000:3000 -d 169942020521.dkr.ecr.eu-west-1.amazonaws.com/local/suppression-web
```

File: src/app.ts (hunk 1 above)
Comment (akaJSaunders) [reply src/app.ts:27]: I left it here for now while we don't have a standardised place for keeping these values

File: src/views/index.njk (hunk 2 above)
Comment (akaJSaunders) [reply src/views/index.njk:6]: That was naughty of me

**Pull request merged**